### DISCUSSION TITLE
Buffered mode to reduce CPU usage?

### DISCUSSION THREAD

**Discussion by 38github** [mikeoliphant/neural-amp-modeler-lv2#68] [Ideas]
I have heard from users of MOD devices that their NAM plugin has a buffered mode that reduces CPU a lot but of course adds latency. Is this something that can be added to your plugin and if so would like 48 samples (1ms) reduce the CPU a lot?

User abotte over at MOD forum reports that if he uses buffered mode the CPU usage of a NANO model goes from 78% to 18%.

**Comment by mikeoliphant** [comment]
I think this uses a background thread to offload the plugin's processing to a second CPU core at the expense of a full buffer worth of latency.

Something like this can potentially be a useful tool, but I think it is better implemented at the host level, like MOD's "portal":

https://forum.mod.audio/t/introducing-portal/9329

**Reply by 38github** [reply]
Thank you for mentioning Portal. I will try that out!
